src/components/pages/vendor.tsx

```diff
@@ -24,6 +25,8 @@ export default function VendorPage() {
     },
   )
 
+  useErrorToast(error)
```

Comment:
This might confuse some people into thinking that the error toast is always shown. Also, if we want to show success toast we have to do something like this,


```
useSuccessToast()
useErrorToast(error)
```

I think we should be able to reengineer this

Reply:
There will be no such thing as `useSuccessToast()` a success response will be extracted from an API response
```ts
const someFn = async () => {
  try {
    const { someData } = await someMutation().unwrap()
    toast({
      // We'll handle success response here
    })
  } catch {
    // Already handled above using `useErrorToast`
  }
}

Feel free to change the name of `useErrorToast` though

Reply:
Basically `useErrorToast` is a wrapper function for handling multiple error scenarios right? I think it would be better to put the toast handler inside the catch block because it follows the pattern.

I think calling it `handleErrorToast` would make it easier to understand, but I'm not sure if we can put it in the catch block because it wraps a hook. Also, do we even need to use `useEffect` though?